tensorflow_federated/python/core/impl/executors/trusted_aggregating_executor.py

```diff
@@ -0,0 +1,446 @@
+      even if the value is of an `all_equal` type or there's only a single
+      participant associated with the given placement.
+
+    * A single value embedded in a subordinate executor in case `type_spec` is
+      of a non-federated non-functional type.
+
+    * An instance of `anonymous_tuple.AnonymousTuple` with values being one of
+      the supported types listed above.
+
+    This constructor does not perform any verification, however.
+
+    Args:
+      value: An internal value representation (of one of the allowed types, as
+        defined above).
+      type_spec: An instance of `tff.Type` or something convertible to it that
+        is compatible with `value` (as defined above).
+    """
+    self._value = value
+    self._type_signature = computation_types.to_type(type_spec)
+    py_typecheck.check_type(type_spec, computation_types.Type)
+
+  @property
+  def internal_representation(self):
+    return self._value
+
+  @property
+  def type_signature(self):
+    return self._type_signature
+
+  @tracing.trace
+  async def compute(self):
+    if isinstance(self._value, executor_value_base.ExecutorValue):
+      return await self._value.compute()
+    elif isinstance(self._type_signature, computation_types.FederatedType):
+      return await self._value[0].compute()
+    elif isinstance(self._value, anonymous_tuple.AnonymousTuple):
+      gathered_values = await asyncio.gather(*[
+          TrustedAggregatingExecutorValue(v, t).compute()
+          for v, t in zip(self._value, self._type_signature)
+      ])
+      type_elements_iter = anonymous_tuple.iter_elements(self._type_signature)
+      return anonymous_tuple.AnonymousTuple(
+          (k, v) for (k, _), v in zip(type_elements_iter, gathered_values))
+    else:
+      raise RuntimeError(
+          'Computing values of type {} represented as {} is not supported in '
+          'this executor.'.format(self._type_signature,
+                                  py_typecheck.type_string(type(self._value))))
+
+class TrustedAggregatingExecutor(executor_base.Executor):
+
+  def __init__(self, target_executors):
+    """Creates a trusted executor for Aggregating.
+
+    Args:
+      target_executors: A dictionary mapping placements to executors or lists of
+        executors associated with these placements. The keys in this dictionary
+        can be either placement literals, or `None` to specify the executor for
+        unplaced computations. The values can be either single executors (if
+        there only is a single participant associated with that placement, as
+        would typically be the case with `tff.SERVER`) or lists of target
+        executors. This dictionary must contain an 'AGGREGATOR' key-value pair
+        targeting the aggregating executor stack.
+
+    Raises:
+      ValueError: If the value is unrecognized (e.g., a nonexistent intrinsic).
+    """
+    assert AGGREGATOR in target_executors
```

Comment:
I meant to make the Aggregator class internal (so that AGGREGATOR was the only user-facing aspect of it). Making it a PlacementLiteral is definitely preferred.